### PR TITLE
add doc for the Huawei OBS new bindings component

### PR DESCRIPTION
**Pull request by jasonlica**
Signed-off-by: jasonlica <jasonli@huawei.com>

Thank you for helping make the Dapr documentation better!

**Please follow this checklist before submitting:**
- [x] Commits are signed with Developer Certificate of Origin (DCO - [learn more](https://docs.dapr.io/contributing/contributing-overview/#developer-certificate-of-origin-signing-your-work))
- [x] [Read the contribution guide](https://docs.dapr.io/contributing/contributing-docs/)
- [x] Commands include options for Linux, MacOS, and Windows within codetabs
- [x] New file and folder names are globally unique
- [x] Page references use shortcodes instead of markdown or URL links
- [x] Images use HTML style and have alternative text
- [x] Places where multiple code/command options are given have codetabs

In addition, please fill out the following to help reviewers understand this pull request:

## Description

Drafted a documentation for the new Dapr bindings component implemented for the Huawei OBS.

## Issue reference

Please reference the issue this PR will close: https://github.com/dapr/docs/issues/2436

### CODE REVIEW

**Comment by jasonlica** [comment]
Hi @greenie-msft this is the new PR which I created, and could you please review and merge? thanks.

**Comment by yaron2** [comment]
@msfussell please review :)

File: daprdocs/content/en/reference/components-reference/supported-bindings/_index.md [hunk 1]
```diff
@@ -90,3 +90,9 @@ Table captions:
 |------|:----------------:|:-----------------:|--------| --------- | ---------- |
 | [Zeebe Command]({{< ref zeebe-command.md >}})            |    | ✅ | Alpha | v1 | 1.2 |
 | [Zeebe Job Worker]({{< ref zeebe-jobworker.md >}})       | ✅ |    | Alpha | v1 | 1.2 |
+
+### Huawei Cloud
+
+| Name | Input<br>Binding | Output<br>Binding | Status | Component version | Since |
+|------|:----------------:|:-----------------:|--------| --------- | ---------- |
+| [Huawei OBS]({{< ref huawei-obs.md >}})            |    | ✅ | Alpha | v1 | 1.8 |
```
Comment (msfussell) [inline daprdocs/content/en/reference/components-reference/supported-bindings/_index.md]:
```suggestion
| [Huawei Object Storage Service (OBS)]({{< ref huawei-obs.md >}})            |    | ✅ | Alpha | v1 | 1.8 |
```

File: daprdocs/content/en/reference/components-reference/supported-bindings/huawei-obs.md [hunk 2]
```diff
@@ -0,0 +1,278 @@
...
+title: "Huawei OBS binding spec"
+linkTitle: "Huawei OBS"
+description: "Detailed documentation on the Huawei OBS binding component"
+aliases:
+  - "/operations/components/setup-bindings/supported-bindings/huawei-obs/"
+---
+
+## Component format
+
+To setup Huawei OBS (output) binding create a component of type `bindings.huawei.obs`. See [this guide]({{< ref "howto-bindings.md#1-create-a-binding" >}}) on how to create and apply a binding configuration.
```
Comment (msfussell) [inline daprdocs/content/en/reference/components-reference/supported-bindings/huawei-obs.md]:
```suggestion
To setup Huawei Object Storage Service (OBS) (output) binding create a component of type `bindings.huawei.obs`. See [this guide]({{< ref "howto-bindings.md#1-create-a-binding" >}}) on how to create and apply a binding configuration.
```

File: daprdocs/content/en/reference/components-reference/supported-bindings/huawei-obs.md [hunk 3]
```diff
@@ -0,0 +1,278 @@
...
+
+{{< /tabs >}}
+
+#### Response
+
+The response JSON body contains the `statusCode` and the `versionId` fields. The `versionId` will have a value returned only if the bucket versioning is enabled and an empty string otherwise.
+
+### Upload file
+
+To upload a binary file (e.g., _.jpg_, _.zip_), invoke the Huawei OBS binding with a `POST` method and the following JSON body:
```
Comment (msfussell) [inline daprdocs/content/en/reference/components-reference/supported-bindings/huawei-obs.md]:
```suggestion
To upload a binary file (for example, _.jpg_, _.zip_), invoke the Huawei OBS binding with a `POST` method and the following JSON body:
```

File: daprdocs/content/en/reference/components-reference/supported-bindings/huawei-obs.md [hunk 4]
```diff
@@ -0,0 +1,278 @@
...
+
+> Note: by default, a random UUID is generated, if we don't specify the `key`. See the example below for Metadata support to set the destination file name. This API can still be used to upload a regular file (i.e., plain text).
```
Comment (msfussell) [inline daprdocs/content/en/reference/components-reference/supported-bindings/huawei-obs.md]:
```suggestion
> Note: by default, a random UUID is generated, if you don't specify the `key`. See the example below for metadata support to set the destination file name. This API can be used to upload a regular file, such as a plain text file).
```

File: daprdocs/content/en/reference/components-reference/supported-bindings/huawei-obs.md [hunk 5]
```diff
@@ -0,0 +1,278 @@
...
+  curl -d '{ "operation": "delete", "metadata": { "key": "my-test-file.txt" }}' \
+        http://localhost:<dapr-port>/v1.0/bindings/<binding-name>
+  ```
+  {{% /codetab %}}
+
+{{< /tabs >}}
+
+#### Response
+
+An HTTP 204 (No Content) and empty body will be retuned if successful.
```
Comment (msfussell) [inline daprdocs/content/en/reference/components-reference/supported-bindings/huawei-obs.md]:
```suggestion
An HTTP 204 (No Content) and empty body are returned if successful.
```

File: daprdocs/content/en/reference/components-reference/supported-bindings/huawei-obs.md [hunk 6]
```diff
@@ -0,0 +1,278 @@
...
+}
+```
+
+The data parameters are:
+
+- `maxResults` - (optional) sets the maximum number of keys returned in the response. By default the action returns up to 1,000 key names. The response might contain fewer keys but will never contain more.
+- `prefix` - (optional) limits the response to keys that begin with the specified prefix.
+- `marker` - (optional) marker is where you want Huawei OBS to start listing from. Huawei OBS starts listing after this specified key. Marker can be any key in the bucket. The marker value may then be used in a subsequent call to request the next set of list items.
+- `delimiter` - (optional) A delimiter is a character you use to group keys. It returns objects/files with their object key other than that is specified by the delimiter pattern.
+
```
Comment (msfussell) [inline daprdocs/content/en/reference/components-reference/supported-bindings/huawei-obs.md:267]: Do you want to provide an example here of the `list` command?

**Comment by msfussell** [comment]
@jasonlica - Looks great! mostly minor updates to style and grammar.

File: daprdocs/content/en/reference/components-reference/supported-bindings/huawei-obs.md (hunk 6 above)
Comment (jasonlica) [reply daprdocs/content/en/reference/components-reference/supported-bindings/huawei-obs.md:267]: Sure, I will add an example for this.

**Comment by jasonlica** [comment]
> @jasonlica - Looks great! mostly minor updates to style and grammar.

Thanks @msfussell for the review and suggestions.

File: daprdocs/content/en/reference/components-reference/supported-bindings/huawei-obs.md (hunk 6 above)
Comment (jasonlica) [reply daprdocs/content/en/reference/components-reference/supported-bindings/huawei-obs.md:267]: @msfussell Added example. pls review, thanks.

**Review by berndverst — APPROVED** [review]
LGTM

**Review by msfussell — APPROVED** [review]
Fantastic Contribution. Appreciate this.

**Comment by jasonlica** [comment]
Thanks @msfussell and @berndverst. It was a pleasure working with the community!

**Pull request merged**